daemon/daemon.go

```diff
@@ -973,15 +974,17 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
 		return nil, err
 	}
 
-	uuid, err := loadOrCreateUUID(filepath.Join(config.Root, "engine_uuid"))
-	if err != nil {
-		return nil, err
-	}
-
-	trustDir := filepath.Join(config.Root, "trust")
-
-	if err := system.MkdirAll(trustDir, 0700, ""); err != nil {
-		return nil, err
+	trustKey, err := libtrust.LoadKeyFile(config.TrustKeyPath)
+	switch {
+	case err == nil:
+		d.ID = trustKey.PublicKey().KeyID()
```

Comment:
Can we write the ID to the new file in this case? That way we're able to preserve the ID once we really remove this logic.

FWIW, the format of the ID is not formalized (the new code uses an UUID for convenience), so storing the old ID format in the new location should not be a problem

Reply:
I'll update

Reply:
@thaJeztah @justincormack the only problem I see here is that we would put a non-UUID into a file called `engine_uuid`. Should we rename the file to `engine_id` ?

Reply:
Makes sense to me; it's used to store an ID (and we describe it in the API that the format is not specified)